packages/playground/blueprints/src/lib/steps/activate-plugin.ts

```diff
@@ -33,7 +39,40 @@ export const activatePlugin: StepHandler<ActivatePluginStep> = async (
 	await playground.run({
 		code: `<?php
       ${requiredFiles.map((file) => `require_once( '${file}' );`).join('\n')}
-      activate_plugin('${pluginPath}');
+      activate_plugin('${pluginRelativePath}');
       `,
 	});
 };
+
+/**
+ * Find plugin entry file from its folder path.
+ *
+ * Based on logic in WordPress class Plugin_Upgrader and function get_file_data.
+ *
+ * @param playground The playground client.
+ * @param pluginFolderPath The plugin folder path.
+ */
+export const findPluginEntryFile = async (
+	playground: UniversalPHP,
+	pluginFolderPath: string
+): Promise<string | undefined> => {
+	const result = await playground.run({
+		code: `<?php
+$files = glob( "${pluginFolderPath}/" . '*.php' );
+if ( $files ) {
+	foreach ( $files as $file ) {
+
+		// Pull only the first 8 KB of the file in.
+		$file_data = file_get_contents( $file, false, null, 0, 8192 );
+		if ( false === $file_data ) continue;
+
+		if ( preg_match( '/^(?:[ \\t]*<\\?php)?[ \\t\\/*#@]*' . preg_quote( 'Plugin Name', '/' ) . ':(.*)$/mi', $file_data, $match ) && $match[1] ) {
+			echo $file;
+			return;
+		}
+	}
```

Comment:
is this already in a WP context? can we call `get_plugin_data()` directly?

Reply:
OK, that's what it does now:

- activatePlugin step: Accept path to plugin directory and find plugin entry file as needed [30d5c9a](https://github.com/WordPress/wordpress-playground/pull/427/commits/30d5c9a9e8d682ba9135e4ee2902319ff51aabcd)

Glad to be rid of that gnarly regex pattern, and let WordPress find the plugin entry file with its internal logic.